File: server/src/main/java/bit/project/server/entity/Advancedpayment.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import java.time.LocalDate;
import javax.persistence.Id;
import java.math.BigDecimal;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Advancedpayment{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDate date;

    private BigDecimal amount;

    @Lob
    private String reason;

    private LocalDateTime tocreation;


    @ManyToOne
    private Employee employee;

    @ManyToOne
    private Advancedpaymentstatus advancedpaymentstatus;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    @JsonIgnore
    @ManyToMany(mappedBy = "advancedpaymentList")
    private List<Salary> salaryList;


    public Advancedpayment(Integer id){
        this.id = id;
    }

    public Advancedpayment(Integer id, String code){
        this.id = id;
        this.code = code;
    }

}

File: client/src/app/entities/advancedpayment.ts
import {User} from './user';
import {Employee} from './employee';
import {DataPage} from '../shared/data-page';
import {Advancedpaymentstatus} from './advancedpaymentstatus';

export class Advancedpayment {
  id: number;
  code: string;
  employee: Employee;
  advancedpaymentstatus: Advancedpaymentstatus;
  date: string;
  amount: number;
  reason: string;
  creator: User;
  tocreation: string;
}

export class AdvancedpaymentDataPage extends DataPage{
    content: Advancedpayment[];
}
